risc0/zkp/rust/src/core/mod.rs

```diff
@@ -42,13 +48,27 @@ pub mod sha_cpu;
 pub mod sha_rng;
 
 /// For x = (1 << po2), given x, find po2.
+/// # Example
+/// ```
+/// # use risc0_zkp::core::to_po2;
+/// #
+/// assert_eq!(to_po2(7), 2);
+/// assert_eq!(to_po2(10), 3);
+/// ```
 pub fn to_po2(x: usize) -> usize {
     (31 - (x as u32).leading_zeros()) as usize
 }
 
 /// Compute `ceil(log_2(value))`
 ///
-/// Find the smallest value `result` such that `2^result >= value`.
+/// Find the smallest `result` such that, for the provided value,
+/// `2^result >= value`.
+/// # Example
+/// ```
+/// # use risc0_zkp::core::log2_ceil;
+/// #
+/// assert_eq!(log2_ceil())
```

Comment:
Does this compile?

Oh, we should probably get rustdoc added to CI!

Reply:
oh absolutely it does not compile, mistakes were made. fixing it up now

Reply:
and yes definitely, i'll follow up on that tomorrow, i am hoping to maximize the number of examples that do double duty as little tests :) clearly i have failed to use this strategy to stop myself

Reply:
The docs build locally even if I `assert_eq!(3, 500000);`. In fact, I hadn't noticed the error because for the most part, I was building and debugging each individual doctest, and just using `cargo doc --open` to visually inspect. This bad example came from a combination of not running and not seeing, so I'm glad our CI caught it. Next time I'll run `cargo test --doc`.

It does look like we're running the tests as part of the CI process:
```
Doc-tests risc0-zkp
...
failures:
6 | assert_eq!(log2_ceil())
  |                       ^ missing tokens in macro arguments
...
test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.75s
```
But this isn't a full rustdoc build, which seems useful; I'll follow up on it.